features/telemetry/protocols/sflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/sflow/parser/proto/flows/Address.java

```diff
@@ -80,22 +82,36 @@ public String toString() {
                 .toString();
     }
 
-    public void writeBson(final BsonWriter bsonWriter) {
+    public void writeBson(final BsonWriter bsonWriter, final SampleDatagramEnrichment enr) {
         bsonWriter.writeStartDocument();
 
         switch (this.type) {
             case IP_V4:
                 bsonWriter.writeName("ipv4");
-                this.ipV4.writeBson(bsonWriter);
+                this.ipV4.writeBson(bsonWriter, enr);
                 break;
             case IP_V6:
                 bsonWriter.writeName("ipv6");
-                this.ipV6.writeBson(bsonWriter);
+                this.ipV6.writeBson(bsonWriter, enr);
                 break;
             default:
                 throw new IllegalStateException();
         }
 
         bsonWriter.writeEndDocument();
     }
+
```

Comment:
If we have the visitor on SFlow elements, we should use it to do the serialization to BSON using the visitor. This will also avoid to pass around the `SampleDatagramEnrichment` as it can be a field in the visitor implementation.

Reply:
Makes sense. I don't comfortable refactoring all of that sFlow packet handling though, lets defer this for another issue?

Reply:
Works for me. Tracking issue: https://issues.opennms.org/browse/NMS-12193.